plugins/modules/sonic_image_management.py

```diff
@@ -0,0 +1,429 @@
+      path: 'file://home/admin/sonic.bin'
+
+- name: Get image installation status
+  dellemc.enterprise_sonic.sonic_image_management:
+    image:
+      command: get-status
+
+- name: Get list of installed images
+  dellemc.enterprise_sonic.sonic_image_management:
+    image:
+      command: get-list
+
+- name: Stage a firmware updater
+  dellemc.enterprise_sonic.sonic_image_management:
+    firmware:
+      command: install
+      path: 'file://home/admin/onie-update-full.bin'
+
+"""
+
+RETURN = """
+status:
+  description: Status of the operation performed.
+  returned: when I(command) is not C(get-status), C(get-list) and C(get-history)
+  type: str
+  sample: SUCCESS
+info:
+  description: Details returned by the specified get operation.
+  returned: when I(command=get-status) or I(command=get-list) or I(command=get-history)
+  type: dict
+  sample: >
+    {
+            "file-download-speed" : "106200",
+            "file-progress" : 100,
+            "file-size" : "1304997870",
+            "file-transfer-bytes" : "1304997870",
+            "install-end-time" : "1695714740",
+            "install-start-time" : "1695714698",
+            "install-status" : "INSTALL_STATE_SUCCESS",
+            "install-status-detail" : "Image install success",
+            "operation-status" : "GLOBAL_STATE_SUCCESS",
+            "transfer-end-time" : "1695714669",
+            "transfer-start-time" : "1695714657",
+            "transfer-status" : "TRANSFER_STATE_SUCCESS",
+            "transfer-status-detail" : "DOWNLOADING IMAGE"
+    }
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.connection import ConnectionError
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.sonic import (
+    to_request,
+    edit_config
+)
+
+
+def validate_and_retrieve_params(module):
+    """Validates the module parameters"""
+    params = {}
+    for category in ('image', 'patch', 'firmware'):
+        if module.params.get(category) and module.params[category].get('command'):
+            if params.get('category') is None:
+                params['category'] = category
+                params.update(module.params[category])
+            else:
+                module.fail_json(msg="Only one image management operation can be performed at a time")
+
+    if module.check_mode and not params['command'].startswith('get-'):
+        module.fail_json(msg='Only get commands are supported while using check mode, but {0} was provided'.format(params['command']))
+
+    if params['command'] == 'install':
+        if not params.get('path'):
+            module.fail_json(msg="{0} -> path is required when {0} -> command = install".format(params['category']))
+    elif params['command'] in ('remove', 'set-default', 'rollback'):
+        if not params.get('name'):
+            module.fail_json(msg="{0} -> name is required when {0} -> command = {1}".format(params['category'], params['command']))
```

Comment:
Here validate missing arguments, should the validation be done for unwanted extra auguments?

Reply:
Have added warning for ignored options.

Reply:
Thanks